server/protocols/jmap/src/main/java/org/apache/james/jmap/http/XUserAuthenticationStrategy.java

```diff
@@ -69,4 +100,18 @@ private Mono<MailboxSession> createMailboxSession(Username username) {
     public AuthenticationChallenge correspondingChallenge() {
         return X_USER_CHALLENGE;
     }
+
+    private Function<HttpServerRequest, Optional<Username>> createUsernameExtractorWithoutSecretValidation() {
+        return httpRequest -> Optional.ofNullable(httpRequest.requestHeaders().get(X_USER_HEADER_NAME))
+            .map(String::trim)
+            .filter(s -> !s.isEmpty())
+            .map(Username::of);
+    }
+
+    private Function<HttpServerRequest, Optional<Username>> createUsernameExtractorWithSecretValidation(String secret) {
+        return httpRequest -> createUsernameExtractorWithoutSecretValidation().apply(httpRequest)
+            .filter(username -> Optional.ofNullable(httpRequest.requestHeaders().get(X_USER_SECRET_HEADER_NAME))
+                .map(secret::equals)
+                .orElse(false));
+    }
```

Comment:
With this code
we don't need one more class `XUserWithSecretAuthenticationStrategy` (it will quiet verbose while it is simple logic)
This code still don't any breaking changes.

Reply:
+1